enhancements/installer/agent-late-binding.md

```diff
@@ -0,0 +1,179 @@
+performed by a different persona.  The Discovery ISO is created by an
+infrastructure admin, who then boots the hosts.  Each host becomes visible to
+the cluster creators but not associated with any specific cluster resource.  At
+a later time, a cluster creator can create or expand a cluster using available
+hosts.
+
+## Motivation
+
+The benefits of late binding are as follows:
+- Adjusts the flow to suit the two different personas (infrastructure
+  administrator and cloud creator).
+- Allows for a more cloud-like use case where compute resources of different
+  flavors can be consumed, similar to IBM Satellite.
+- Allows agent-based installation to fit more naturally into a MAPI or CAPI
+  model.  While the details of such an integration are not in the scope of
+  this enhancement, the idea is that hosts are booted, thus creating a
+  collection of unassigned agents, and the machine creation involves choosing one
+  such agent and beginning its installation.
+- Reduces the number of Discovery ISOs, as there is no need to generate and
+  download one for each cluster installation.  Each ISO incurs capacity
+  overhead for the service, and management overhead for the user which conflicts
+  with our ease-of-use goal.
+- Allows the user to see their inventory and then make decisions on how to
+  build clusters based on hardware and connectivity.  At a later stage, the
+  service can make recommendations.  This would be a layer of logic above late
+  binding and is out of scope for this enhancement.
+- Enables the user story where the Discovery ISO is loaded at the factory, and
+  then discovered at a later time.
+
+### Goals
+
+- Make the InfraEnv and Agent resources independent from the ClusterDeployment.
+  A single InfraEnv should have the potential to boot any number of hosts,
+  whose Agents register with no relation to any particular ClusterDeployment.  An
+  Agent may be assigned to a ClusterDeployment at a later time.
+- An Agent may be associated with a different ClusterDeployment at any time.
+
+### Non-Goals
+
+- Once a host boots from disk, booting the Discovery ISO again to return it to
+  the collection of free Agents is considered to be a manual step.  There are
+  several options that we will explore whereby this can be done automatically,
+  but that discussion is out of scope for this enhancement.
+- At a later stage, Agents may be automatically assigned to clusters based on
+  their discovered inventory and/or network configurations. This is future work
+  and not in the scope of this enhancement.
+
+## Proposal
+
+The first major change is related to the Discovery ISO.  The only API change
+(which is non-breaking) is that ClusterRef will no longer mandatory.  Once an
+agent registers with the service, its corresponding Agent CR must be created in
+the namespace of the InfraEnv which generated its Discovery ISO.  The
+controller currently relies on the fact that an agent is always associated with
```

Comment:
This feels like a topic for the implementation details section. How we accomplish this is not central to the proposal, and moving this would help keep the "Proposal" section more focused.